### PR TITLE
Dropdown component, list items

### PR DESCRIPTION
**Pull request by MelSumner**
## :pushpin: Summary

If merged, this PR would add the dropdown items to the dropdown component branch.

## :camera_flash: Screenshots

Current state:
 image

## :link: External links

- (internal) [link to CRD](https://docs.google.com/document/d/1PSIAdUypvGWCLBAkeOo4HSa3qd6qN-pWvlLrfuwYT4o/edit#heading=h.gjdgxs)
- (internal) [link to Figma](https://www.figma.com/file/noyY6dUMDYjmySpHcMjhkN/branch/DUjKDINH2ps9ggk6YsAVC6/HDS-Product---Components-%5BWIP%5D?node-id=4861%3A18472)

***

## :white_check_mark: Reviewer's checklist

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

**Comment by MelSumner** [comment]
Note: increase padding or margin _instead_ of line-height, because I don't want multi-line text to be effected by the changes.

**Comment by MelSumner** [comment]
ok @didoo I think this one is ready to roll up into the `dropdown/component` branch with a squash & merge to tidy it all up.

File: packages/components/addon/components/hds/dropdown/toggle-user.hbs [hunk 1]
```diff
@@ -1,11 +1,11 @@
 <button
   class={{this.classNames}}
-  ...attributes
   aria-label={{@text}}
   type="button"
   disabled={{if this.isDisabled "disabled" null}}
+  ...attributes
```
Comment (didoo) [inline packages/components/addon/components/hds/dropdown/toggle-user.hbs:6]: [chore] you don't need attributes spread here, if you look at how it's used in the `index` there are not attributes applied: https://github.com/hashicorp/design-system/blob/44d75fa7d74b9a6a85a6924a79a224f3cd93303f/packages/components/addon/components/hds/dropdown/index.hbs#L4-L9

File: packages/components/addon/components/hds/dropdown/toggle-user.hbs [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
 >
-  <div class="hds-dropdown-toggle__user">
+  <div class="hds-dropdown-toggle-user--avatar">
```
Comment (didoo) [inline packages/components/addon/components/hds/dropdown/toggle-user.hbs:8]: [chore] this is not a _modifier_, is an [_element_](http://getbem.com/naming/), so should be `hds-dropdown-toggle-user__avatar`

File: packages/components/addon/components/hds/dropdown/toggle-user.js [hunk 3]
```diff
@@ -34,7 +34,7 @@ export default class HdsDropdownToggleUserComponent extends Component {
    * @return {string} The "class" attribute to apply to the component.
    */
   get classNames() {
-    let classes = ['hds-dropdown-toggle', 'hds-dropdown-toggle--user'];
+    let classes = ['hds-dropdown-toggle', 'hds-dropdown-toggle-user'];
```
Comment (didoo) [inline packages/components/addon/components/hds/dropdown/toggle-user.js:37]: [chore] same as above, if we use define this as a block, then the `hds-dropdown-toggle` should be removed.

File: packages/components/app/styles/components/dropdown.scss [hunk 4]
```diff
@@ -18,27 +14,24 @@
 // DROPDOWN COMPONENT
 //
 // properties within each class are sorted alphabetically
-// notice: pseudo-classes for the states *must* follow the order link > visited > focus > hover > active
+// notice: pseudo-classes for the states *must* follow the order link > visited > hover > focus > active
```
Comment (didoo) [inline packages/components/app/styles/components/dropdown.scss:17]: [chore]: in all the other components we're saying something different. if we change it here, let's change it across all the other components

File: packages/components/app/styles/components/dropdown.scss [hunk 5]
```diff
@@ -110,36 +111,41 @@ $hds-dropdown-list-item-border-radius: 5px;
...
 
-
-// MORE
+// TOGGLE/MORE
 
 // these values later may come from the design tokens
 $size-props: (
-  "small": (
-    "size": 28px
+  'small': (
```
Comment (didoo) [inline packages/components/app/styles/components/dropdown.scss:118]: [question] in all the other components we're using double quotes in Scss, why these changes in this file?

File: packages/components/app/styles/components/dropdown.scss [hunk 6]
```diff
@@ -110,36 +111,41 @@ $hds-dropdown-list-item-border-radius: 5px;
...
+      height: map-get($size-props, $size, 'size');
+      width: map-get($size-props, $size, 'size');
     }
   }
+
+  &:active,
+  &.is-active {
+    background-color: var(--token-color-surface-interactive-active);
+    border-color: var(--token-color-border-strong);
+  }
```
Comment (didoo) [inline packages/components/app/styles/components/dropdown.scss:138]: [chore] I suspect you don't need these, it's already declared in the top part of the file (lines 96-100). can you check?

File: packages/components/app/styles/components/dropdown.scss [hunk 7]
```diff
@@ -110,36 +111,41 @@ $hds-dropdown-list-item-border-radius: 5px;
...
 
-// USER
 
-.hds-dropdown-toggle--user {
+.hds-dropdown-toggle-user {
   height: $hds-dropdown-toggle-base-height;
   padding: 1px;
 
-  .hds-dropdown-toggle__user {
+  .hds-dropdown-toggle-user--avatar {
```
Comment (didoo) [inline packages/components/app/styles/components/dropdown.scss:148]: see comment above: it's an element, so `.hds-dropdown-toggle-user__avatar`

File: packages/components/app/styles/components/dropdown.scss [hunk 8]
```diff
@@ -156,171 +162,208 @@ $size-props: (
...
-  box-shadow: 0px 0px 0px 1px rgba(101, 106, 118, 0.25), 0px 2px 3px rgba(101, 106, 118, 0.15), 0px 16px 16px -10px rgba(101, 106, 118, 0.2);
-  box-sizing: border-box;
-  display: flex;
-  flex-direction: column;
+  box-shadow: var(--token-surface-high-box-shadow);
+  box-sizing: border-box;  //TODO 
   list-style: none;
-  margin: 0;
-  min-height: fit-content;
+  margin: 6px 0 0;
```
Comment (didoo) [inline packages/components/app/styles/components/dropdown.scss:175]: [question]: why this margin? (if it's to separate the element from the toggle, this should be applied as `top` value to the absolute definitions below)

File: packages/components/app/styles/components/dropdown.scss [hunk 9]
```diff
@@ -156,171 +162,208 @@ $size-props: (
...
-    min-height: 36px;
-    justify-content: center;
+
+// LIST > LIST-ITEM
+// HDS::DROPDOWN::LIST-ITEM
+
+.hds-dropdown-list-item {
+  // TODO flex removed? (MS)
+
+  &.hds-dropdown-list-item--heading {
```
Comment (didoo) [inline packages/components/app/styles/components/dropdown.scss:199]: [issue] you don't need to nest these declarations, they're are already name-spaced, so there's no risk of naming conflicts (plus, the specificity increases)

File: packages/components/app/styles/components/dropdown.scss [hunk 10]
```diff
@@ -156,171 +162,208 @@ $size-props: (
...
+      // remove the focus ring on "active + focused" state (by design) // TODO Mel fix this
+      &:focus:active,
+      &.is-focus.is-active {
+        &::after {
+          box-shadow: none;
+        }
+      }
+    }
+  
+    &.hds-dropdown-list-item--link--action {
```
Comment (didoo) [inline packages/components/app/styles/components/dropdown.scss]: [issue]: I think there must been some conflicts which have not been merged correctly, here the code was different, now it doesn't work anymore. if you want to see how it was done, have a look at this commit: https://github.com/hashicorp/design-system/pull/68/commits/cf72582efb2a5bf4d3421aa4d148cfcbbdb406d2

File: packages/components/app/styles/components/dropdown.scss [hunk 11]
```diff
@@ -156,171 +162,194 @@ $size-props: (
...
+      // remove the focus ring on "active + focused" state (by design) // TODO Mel fix this
+      &:focus:active,
+      &.is-focus.is-active {
+        &::after {
+          box-shadow: none;
+        }
+      }
+    }
+  
+    &.hds-dropdown-list-item--link--color-action {
```
Comment (didoo) [inline packages/components/app/styles/components/dropdown.scss:306]: [issue] I think one of the conflicts fixes after a merge have removed a big chunk of code, now all the styling is gone. you can see how it was done before in this commit: https://github.com/hashicorp/design-system/pull/68/commits/cf72582efb2a5bf4d3421aa4d148cfcbbdb406d2

**Review by didoo — COMMENTED** [review]
@MelSumner I have added a bunch of comments. the most critical one is that some merge/conflict was not fixed in the right way and a bunch of code have been lost (the styling of the links for `action/critical` with all the states)

File: packages/components/app/styles/components/dropdown.scss (hunk 5 above)
Comment (MelSumner) [reply packages/components/app/styles/components/dropdown.scss:118]: It's probably stylelint. I can undo it (I did think single quotes were correct for CSS though)

Comment (didoo) [reply packages/components/app/styles/components/dropdown.scss:118]: they're both correct, only most of the files are using double quotes

File: packages/components/addon/components/hds/dropdown/toggle-user.hbs (hunk 2 above)
Comment (MelSumner) [reply packages/components/addon/components/hds/dropdown/toggle-user.hbs:8]: it's a "type" of toggle/user...but tbh I'm getting a little mixed up here so I'm drawing up a visualization for us to review.

File: packages/components/app/styles/components/dropdown.scss (hunk 4 above)
Comment (MelSumner) [reply packages/components/app/styles/components/dropdown.scss:17]: "lord vader hates fluffy animals" is the mnemonic device I use, it's possible I mixed it up before, noticed it today.

File: packages/components/addon/components/hds/dropdown/toggle-overflow.js [hunk 12]
```diff
@@ -66,10 +66,10 @@ export default class HdsDropdownToggleMoreComponent extends Component {
    * @return {string} The "class" attribute to apply to the component.
    */
   get classNames() {
-    let classes = ['hds-dropdown-toggle', 'hds-dropdown-toggle--more'];
+    let classes = ['hds-dropdown-toggle', 'hds-dropdown-toggle-overflow'];
```
[the remark this answers is not in the document]
Comment (MelSumner) [reply packages/components/addon/components/hds/dropdown/toggle-overflow.js:69]: I mean it shouldn't have been that way, but then it got all jumbled up when the single toggle file was turned into six different toggle files. It's been pretty harsh to clean up TBH.

[the remark this answers is not in the document]
Comment (didoo) [reply packages/components/addon/components/hds/dropdown/toggle-overflow.js]: done

File: packages/components/addon/components/hds/dropdown/toggle-user.js (hunk 3 above)
Comment (didoo) [reply packages/components/addon/components/hds/dropdown/toggle-user.js]: done

File: packages/components/app/styles/components/dropdown.scss (hunk 6 above)
Comment (didoo) [reply packages/components/app/styles/components/dropdown.scss]: removed

File: packages/components/app/styles/components/dropdown.scss (hunk 8 above)
Comment (didoo) [reply packages/components/app/styles/components/dropdown.scss]: fixed

File: packages/components/app/styles/components/dropdown.scss (hunk 10 above)
Comment (didoo) [reply packages/components/app/styles/components/dropdown.scss]: fixed the conflicts

**Review by didoo — APPROVED** [review]
Pair programmed with @MelSumner and fixed all the issues together

File: packages/components/addon/components/hds/dropdown/toggle-user.hbs (hunk 1 above)
Comment (didoo) [reply packages/components/addon/components/hds/dropdown/toggle-user.hbs:6]: we'll make a decision on this later, we may want to use a different pattern (pass the components as yielded elements in the block)

File: packages/components/app/styles/components/dropdown.scss (hunk 7 above)
Comment (didoo) [reply packages/components/app/styles/components/dropdown.scss]: done

File: packages/components/app/styles/components/dropdown.scss (hunk 9 above)
Comment (didoo) [reply packages/components/app/styles/components/dropdown.scss]: done
